### DISCUSSION TITLE
what is the Key Value pair to define Replication Factor in the json object

### DISCUSSION THREAD

**Discussion by Kannadasa** [ravendb/ravendb#14663] [Q&A]
-d '{\"Database\":\"DB_Kanna6\",  \"ReplicationFactor\":\"3\"}'

The replication factor key value pair name seems to be wrong and CURL command ignores this value while creating a database and sets the replication factor to 1.

Please tell me know the correct Key value name to define replication factor.

Thanks

**Comment by Kannadasa** [comment]
Please ignore the escaping out errors, i am using correct syntax to escape out quotes

**Reply by ayende** [reply]
The replicationFactor is passed as a query string argument, see:
https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Client/ServerWide/Operations/CreateDatabaseOperation.cs#L65

**Reply by Kannadasa** [reply]
-d '{\"DatabaseName\":\"Kanna11\", \"Topology\":{\"ReplicationFactor\": 3}}'

This also does not work

**Reply by Kannadasa** [reply]
> The replicationFactor is passed as a query string argument, see: https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Client/ServerWide/Operations/CreateDatabaseOperation.cs#L65

Thanks Ayende, how can i use this within a curl command.  I am using curl command to create a database.

-d '{\"DatabaseName\": \"DB_Kanna1\", \"ReplicationFactor\": \"2\"}'

curl.exe -X PUT -H "Content-Type: application/json" -d '{\"DatabaseName\": \"DB_Kanna1\", \"ReplicationFactor\": \"2\"}' --cert "LocalMachine\My\xxxx" --key "D:\RavenDB\certificates\xxx.key" https://xxx:000/admin/databases   .

This is the command i am using to create a database and i want to mention the replication factor in the same command.

Thanks
Kannadasan

**Reply by Kannadasa** [reply]
> The replicationFactor is passed as a query string argument, see: https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Client/ServerWide/Operations/CreateDatabaseOperation.cs#L65

I am not using any language to create a database.  I am using CURL commands to create a database with the replication factor.

Thanks
Kannadasan

**Reply by ayende** [reply]
Look at what it is doing, it is passing this as a query string parameter, not inside the json

**Reply by Kannadasa** [reply]
Yes you are correct.  How can i define replication factor inside a Curl command using JSON object? something like below.

PUT /admin/databases

{
    "DatabaseName": "DB_NAME"
    "ReplicationFactor": 3
}

**Reply by Kannadasa** [reply]
Is this not possible to run the below code block using Curl command?

PUT /admin/databases

{
"DatabaseName": "DB_NAME"
"ReplicationFactor": 3
}

**Reply by ayende** [reply]
You do no.

You use:

`PUT /admin/databases?replicationFactor=3`

**Reply by Kannadasa** [reply]
Thanks, it worked
